test/dartdoc_test_base.dart

```diff
@@ -70,23 +72,37 @@ analyzer:
         packagePath, name, Uri.file('$packagePath/'));
   }
 
+  Future<PackageGraph> _bootPackageFromFiles(FileGenerator files) async {
+    var packagePathBasename =
+        resourceProvider.pathContext.basename(packagePath);
+    var packagePathDirname = resourceProvider.pathContext.dirname(packagePath);
+    await d
+        .dir(packagePathBasename, files())
+        .createInMemory(resourceProvider, packagePathDirname);
+    return await bootBasicPackage(
+      packagePath,
+      packageMetaProvider,
+      packageConfigProvider,
+    );
+  }
+
+  /// Creates a single library named [libraryName], with optional preamble
+  /// [libraryPreamble].   Optionally, pass a [FileGenerator] to create
+  /// extra files in the package such as `dartdoc_options.yaml`.
   Future<Library> bootPackageWithLibrary(String libraryContent,
-      {String libraryPreamble = ''}) async {
-    await d.dir('lib', [
-      d.file('lib.dart', '''
+      {String libraryPreamble = '', FileGenerator? extraFiles}) async {
```

Comment:
The new `extraFiles` parameter looks unused?

Reply:
good catch, missed that in a refactor.